geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegionMap.java

```diff
@@ -1625,10 +1629,10 @@ public boolean invalidate(EntryEventImpl event, boolean invokeCallbacks, boolean
                       if (isDebugEnabled) {
                         logger.debug("Invalidate: Entry already invalid: '{}'", event.getKey());
                       }
-                      if (event.getVersionTag() != null && owner.getVersionVector() != null) {
-                        owner.getVersionVector().recordVersion(
-                            (InternalDistributedMember) event.getDistributedMember(),
-                            event.getVersionTag());
+                      processVersionTag(re, event);
```

Comment:
You added this block of code in two places and they are both proceeded by the same logging and call of processVersionTag.
Could all of this code be extracted into a method? It could be named something like "handleAlreadyInvalidEntry(RegionEntry)".

Reply:
sure, I'll extract that to a method.